File: example/src/main/java/com/example/reveallayout/MainActivity.java
package com.example.reveallayout;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.Button;

import butterknife.BindViews;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
public class MainActivity extends BaseActivity implements View.OnClickListener {

    @BindViews({R.id.button_main_simple, R.id.button_main_custom}) Button[] buttons;

    @Override
    public int getContentView() {
        return R.layout.activity_main;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        for (Button button : buttons)
            button.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        startActivity(new Intent(this, v.getId() == R.id.button_main_simple
                ? SimpleActivity.class
                : CustomActivity1.class));
    }
}

File: example/src/main/java/com/example/reveallayout/MainActivity.kt
package com.example.reveallayout

import android.content.Intent
import android.graphics.Rect
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.activity_main.*

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        button.setOnClickListener {
            Intent(this, SecondActivity::class.java).apply {
                addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION)
                putExtra(SecondActivity.EXTRA_RECT, createRect(button))
                startActivity(this)
            }
        }
    }

    private fun createRect(view: View) = Rect().apply {
        view.getDrawingRect(this)
        (view.parent as ViewGroup).offsetDescendantRectToMyCoords(view, this)
    }
}